ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -455,9 +457,15 @@ def process(self, device, results, unused):
 
         # Find all hosts which have a neutron agent on them.
         for agent in results['agents']:
-            host_id = prepId("host-{0}".format(agent['host']))
+            log.debug("Sanitizing hostname %s", agent['host'])
+            sanitized_hostname = sanitize_host_or_ip(agent['host'])
+            if not sanitized_hostname:
+                log.debug("Sanitized hostname %s is empty!", agent['host'])
+                continue
+
+            host_id = prepId("host-{0}".format(sanitized_hostname))
             hostmap[host_id] = {
-                'hostname': agent['host'],
+                'hostname': sanitized_hostname,
```

Comment:
I think you also need to do the same in the call to set the org component for the nova agent- the one that originally threw the error.   Also not sure you've fixed the underlying bug there because it looks like your fix below (good catch, btw) is only going to affect nova api hosts, not hosts with neutron agents on them.